pkg/kubelet/kubelet.go

```diff
@@ -1251,6 +1254,9 @@ func (kl *Kubelet) setupDataDirs() error {
 	if err := os.MkdirAll(kl.getPodResourcesDir(), 0750); err != nil {
 		return fmt.Errorf("error creating podresources directory: %v", err)
 	}
+	if err := os.MkdirAll(kl.getCheckpointsDir(), 0750); err != nil {
```

Comment:
we should only create the checkpoint directory if the feature gate is enabled.

Reply:
Good point. Changed.